stats/teams/new_team.php

```diff
@@ -35,12 +35,19 @@
 }
 else if (isset($_POST['mkMake']))
 {
-    $result = mysqli_query(DPDatabase::get_connection(), sprintf("
+    $teamname = stripAllString(trim($_POST['teamname']));
```

Comment:
If we want to futz with this (and we should -- the team code is horrific) we should move these to the top of the file and always populate them instead of pulling them in various places. We also need to use `array_get()` to pull the variables because your code is generating a notice as-is for lines 41 & 42 when an avatar and icon are not provided.

Reply:
Hopefully this should be better... I don't love the way we accept and silently trim spaces, and strip a subset of markup and escape the rest. Do you think it might be better if we just drop the stripAllString, and validate each piece of input: disallowing any leading or trailing space in the team names, and refusing < > and & in teamname, text_data, teamwebpage?

Reply:
This is better, thanks.

The team code makes me want to claw my eyes out. Per-item validation would be better, as would be dropping our custom formatting (really: why?). But none of this is important enough to work on right now IMHO.